charts/patch/templates/patch.json

```diff
@@ -1,5 +1,38 @@
 {{ $prefix := .Values.pathPrefix -}}
+{{ $alreadyInjected := .Values.alreadyInjected -}}
 [
+  {{- if $alreadyInjected }}
```

Comment:
Can you explain why these annotations need to be removed?  My understanding is that the `add` op will replace the previous value if it exists, so I wouldn't expect it to be a problem if these are not explicitly removed during reinvocation.

Reply:
If an `add` op tries to add something that's already there, it'll generate an error and the pod creation will fail.

Reply:
The jsonpatch spec https://datatracker.ietf.org/doc/html/rfc6902#section-4.1 states:

>   A member to add to an existing object - whereupon the supplied
      value is added to that object at the indicated location.  If the
      member already exists, it is replaced by the specified value.

Very unfortunate if Kubernetes doesn't follow this.

Reply:
You're right. I have read that as applying only to objects, not arrays. Plus not explicitly removing the initContainer nor the proxy container does return an error, which reinforced that misunderstanding. But I think now that's just because those additions _append_ to arrays (using `-` at the end of the path) for which apparently this replacing logic you quoted doesn't apply.
Anyways, I'll remove those annotation removals :+1: